### PR TITLE
firewall: remove PF3/bootstrap dependency

### PR DESCRIPTION
**Pull request by KKoukiou**
The 'Add service to zone' dialog was sligthly refactored to avoid usage
of custom components. The new design makes the UI more user friendly,
since the radio buttons don't jump around any more, when switching
between the two radio button options, and the attached text is revealed.

### CODE REVIEW

**Comment by KKoukiou** [comment]
@garrett I need help with the Trust level field in add zones dialog.

**Comment by garrett** [comment]
I'm looking at this now. It's a great start.

I've noticed a bunch of issues:

1. Heading is the wrong size
2. Radios in the add services dialog are not implemented properly, due to lack of accessibility. This also makes it so you cannot click the labels (which is an easy way to check this, and affects everyone, not just those relying on screen readers).
3. The add services progress indicator has an overflow visible when loading, causing a progress bar to show up and jump around while the list is loading.
4. It's not possible to click the list to select the checkboxes — the target area is too small. At a minimum, the `service-list-iteam-heading`s should be labels associated with the checkboxes. (This is another a11y issue.) Additionally, some JS could allow the entire row act as a click. (But this would be an enhancement. The labels being associated with checkboxes via IDs and/or label wrapping are mandatory regardless.)
5. The radio group in allowed addresses has center alignment? They're also not accessible.
6. Checkboxes for interfaces are not accessible.
7. Odd spacing in the "zone contains Cockpit" modal. Is this meant to be a warning, but the styling is missing? (It should be a warning and use PF warning widgets.)
8. Same issue for other confirmation modals. I think they're supposed to be warnings. ([PF demo](https://www.patternfly.org/v4/components/modal/html#warning-alert), [React implementation](https://www.patternfly.org/v4/components/modal/react#warning-alert))
9. And, of course, the style of the trust level field (which you pointed out). :wink:

I'll work more on this PR tomorrow. (Feel free to pick off anything from the list; we can sync up to figure out who does what.)

I've pushed an example fix for the radio switcher in the services dialog.

**Comment by KKoukiou** [comment]
> I'm looking at this now. It's a great start.
> 
> I've noticed a bunch of issues:
> 
>     * [ ]  Heading is the wrong size
> 
>     * [x]  Radios in the add services dialog are not implemented properly, due to lack of accessibility. This also makes it so you cannot click the labels (which is an easy way to check this, and affects everyone, not just those relying on screen readers).
> 
>     * [ ]  The add services progress indicator has an overflow visible when loading, causing a progress bar to show up and jump around while the list is loading.

Do you mean the spinner?

> 
>     * [x]  It's not possible to click the list to select the checkboxes — the target area is too small. At a minimum, the `service-list-iteam-heading`s should be labels associated with the checkboxes. (This is another a11y issue.) Additionally, some JS could allow the entire row act as a click. (But this would be an enhancement. The labels being associated with checkboxes via IDs and/or label wrapping are mandatory regardless.)
> 
>     * [x]  The radio group in allowed addresses has center alignment? They're also not accessible.
> 
>     * [x]  Checkboxes for interfaces are not accessible.
> 
>     * [x]  Odd spacing in the "zone contains Cockpit" modal. Is this meant to be a warning, but the styling is missing? (It should be a warning and use PF warning widgets.)

I just added an inline alert, because the modals can be warning type, but I don't know where to put the warning message itself in this case.
> 
>     * [x]  Same issue for other confirmation modals. I think they're supposed to be warnings. ([PF demo](https://www.patternfly.org/v4/components/modal/html#warning-alert), [React implementation](https://www.patternfly.org/v4/components/modal/react#warning-alert))
> 
>     * [ ]  And, of course, the style of the trust level field (which you pointed out). wink
> 
> 
> I'll work more on this PR tomorrow. (Feel free to pick off anything from the list; we can sync up to figure out who does what.)
> 
> I've pushed an example fix for the radio switcher in the services dialog.

**Comment by garrett** [comment]
> Do you mean the spinner?

Right, indeterminate progress. I guess progress actually would imply _showing_ the progress instead of "stuff's happening... dunno" like the spinner.

So, yes, spinner. :wink:

> I just added an inline alert, because the modals can be warning type, but I don't know where to put the warning message itself in this case.

I'll look.

FWIW: I'm not so much of a fan of the PF4 alert modals anyway. You can't even tell that they're an alert, except for an icon that can be missable. Perhaps we want to ask them for a redesign and a style suggestion, based on the style of the inline alerts, to make them move obvious?

**Comment by KKoukiou** [comment]
> > Do you mean the spinner?
> 
> Right, indeterminate progress. I guess progress actually would imply _showing_ the progress instead of "stuff's happening... dunno" like the spinner.
> 
> So, yes, spinner. wink
> 
> > I just added an inline alert, because the modals can be warning type, but I don't know where to put the warning message itself in this case.
> 
> I'll look.
> 
> FWIW: I'm not so much of a fan of the PF4 alert modals anyway. You can't even tell that they're an alert, except for an icon that can be missable. Perhaps we want to ask them for a redesign and a style suggestion, based on the style of the inline alerts, to make them move obvious?

Makes sense. I also don't see where the alert warning should go in these modals.

File: pkg/networkmanager/firewall.jsx [hunk 1]
```diff
@@ -701,26 +702,27 @@ class ActivateZoneModal extends React.Component {
...
+                    <FormGroup label={ _("Trust level") } className="add-zone-zones">
+                        <Flex>
+                            <FlexItem className="add-zone-zones-firewalld">
+                                <legend>{ _("Sorted from least to most trusted") }</legend>
                                 { zones.filter(z => firewall.predefinedZones.indexOf(z) !== -1).sort((a, b) => firewall.predefinedZones.indexOf(a) - firewall.predefinedZones.indexOf(b))
                                         .map(z =>
                                             <Radio key={z} id={z} name="zone" value={z}
-                                                   onChange={(value, e) => this.onChange("zone", e.target.value)}
-                                                   label={ firewall.zones[z].id } />
+                                                    defaultChecked={ firewall.predefinedZones.indexOf(z) === 1 }
```
Comment (garrett) [inline pkg/networkmanager/firewall.jsx:712]: I couldn't get this working. I was trying to make the first default. I decided to give up and ask for you to help here.

So, what should this be like?

File: pkg/networkmanager/firewall.jsx [hunk 2]
```diff
@@ -701,26 +702,27 @@ class ActivateZoneModal extends React.Component {
                    </>}
             >
                 <Form isHorizontal>
-                    <FormGroup label={ _("Trust level") }>
-                        <div role="group" className="add-zone-zones">
-                            <div className="add-zone-zones-firewalld">
-                                <legend>{ _("Sorted from least trusted to most trusted") }</legend>
+                    <FormGroup label={ _("Trust level") } className="add-zone-zones">
+                        <Flex>
+                            <FlexItem className="add-zone-zones-firewalld">
```
Comment (garrett) [inline pkg/networkmanager/firewall.jsx:707]: I figured we could just use PF4's Flex here.

File: pkg/networkmanager/firewall.jsx [hunk 3]
```diff
@@ -701,26 +702,27 @@ class ActivateZoneModal extends React.Component {
...
-                            <div className="add-zone-zones-custom">
+                            </FlexItem>
+                            <FlexItem className="add-zone-zones-custom">
                                 { customZones.length > 0 && <legend>{ _("Custom zones") }</legend> }
                                 { customZones.map(z =>
                                     <Radio key={z} id={z} name="zone" value={z}
-                                           onChange={(value, e) => this.onChange("zone", e.target.value)}
-                                           label={ firewall.zones[z].id } />
+                                            onChange={(value, e) => this.onChange("zone", e.target.value)}
+                                            label={ firewall.zones[z].id } />
```
Comment (garrett) [inline pkg/networkmanager/firewall.jsx:722]: Oh weird; did I mess up the indentation? If so, whoops. ESLint didn't complain though.

File: pkg/networkmanager/interfaces.js [hunk 4]
```diff
@@ -41,7 +41,7 @@ import 'bootstrap/dist/js/bootstrap';
 import "page.scss";
 import "table.css";
 import "plot.css";
-import "./networking.css";
+import "./networking.scss";
```
Comment (garrett) [inline pkg/networkmanager/interfaces.js:44]: I needed to rename the file. The contents were SCSS but the filename was CSS still.

File: pkg/networkmanager/networking.scss [hunk 5]
```diff
@@ -415,32 +415,16 @@ h2 + .onoff-ct {
...
+    line-height: 3;
 }
 
-#add-zone-dialog .add-zone-zones label {
-    margin: 0;
-    flex-flow: column;
-    padding: 0 0.5rem;
+#add-zone-dialog .add-zone-zones .pf-c-radio__label {
     text-transform: capitalize;
-    font-size: var(--pf-global--FontSize--sm);
```
Comment (garrett) [inline pkg/networkmanager/networking.scss]: I'm wondering if we could get rid of even more CSS in the file.

File: pkg/networkmanager/firewall.jsx (hunk 1 above)
Comment (KKoukiou) [reply pkg/networkmanager/firewall.jsx:712]: I will repush and make it default

File: pkg/networkmanager/networking.scss (hunk 5 above)
Comment (KKoukiou) [reply pkg/networkmanager/networking.scss]: The more you can remove the better ;)

File: pkg/networkmanager/firewall.jsx (hunk 3 above)
Comment (KKoukiou) [reply pkg/networkmanager/firewall.jsx:722]: I fixed it back manually. You probably have some auto indent feature in your editor?

Comment (garrett) [reply pkg/networkmanager/firewall.jsx]: Yeah, I was indenting / re-indenting a few blocks, and I guess I — or my editor — got it wrong.

File: pkg/networkmanager/networking.scss (hunk 5 above)
Comment (garrett) [reply pkg/networkmanager/networking.scss]: I figure we'll do that in #15054, after this lands. I don't want to accidentally break stuff on a PR we're both happy with right now, on both of our last days this year. We can finish up the cleanup in 2021. 😉

There are blocks I'm 99% certain we can just remove (when the time comes).

**Review by martinpitt — APPROVED** [review]
The code changes look great, and I (lightly) tested this visually in a F33 VM. Nice cleanup, thanks!

**Comment by martinpitt** [comment]
[firewall test failure on fedora-33](https://logs.cockpit-project.org/logs/pull-15046-20210104-103349-34a20e3b-fedora-33/log.html#311). This is probably a flake, but retrying to make sure.

**Comment by KKoukiou** [comment]
> [firewall test failure on fedora-33](https://logs.cockpit-project.org/logs/pull-15046-20210104-103349-34a20e3b-fedora-33/log.html#311). This is probably a flake, but retrying to make sure.

Ah this seems to be flaky, see failure-stats for fedora-33 https://images-frontdoor.apps.ocp.ci.centos.org/failure-stats.html

**Comment by KKoukiou** [comment]
> > [firewall test failure on fedora-33](https://logs.cockpit-project.org/logs/pull-15046-20210104-103349-34a20e3b-fedora-33/log.html#311). This is probably a flake, but retrying to make sure.
> 
> Ah this seems to be flaky, see failure-stats for fedora-33 https://images-frontdoor.apps.ocp.ci.centos.org/failure-stats.html

It also succeeded once in this PR, then the second retry failed (we ran 3 times the 'affected' tests).
I will merge since it's not a new flake, and try to address it separately.

**Pull request merged**